frontend/packages/shared/src/contexts/ServicesContext.tsx

```diff
@@ -39,23 +39,19 @@ const handleError = (
 ): void => {
   // TODO : log axios errors
 
-  if (error?.response?.status === ServerCodes.Unauthorized) {
-    const errorMessageKey = 'api_errors.Unauthorized';
-    if (i18n.exists(errorMessageKey)) {
-      toast.error(t(errorMessageKey), { toastId: errorMessageKey, autoClose: LOG_OUT_TIMER_MS });
-    }
+  const errorCode = error?.response?.data?.errorCode;
+  const errorMessageKey = errorCode ? `api_errors.${errorCode}` : 'api_errors.Unauthorized';
+
+  const renderToast = () => {
+    toast.error(t(errorMessageKey), { toastId: errorMessageKey, autoClose: LOG_OUT_TIMER_MS });
     setTimeout(() => {
       logout().then(() => window.location.assign(userLogoutAfterPath()));
     }, LOG_OUT_TIMER_MS);
```

Comment:
I would suggest moving this code inside the `onClose` function of the toast. This will prevent using a `setTimeout`

Reply:
I moved it but still logout 😢 , i will check it again.